src/rpc/util.h

```diff
@@ -329,19 +329,53 @@ struct RPCExamples {
     std::string ToDescriptionString() const;
 };
 
+template <typename T>
+struct RPCDefaultParam
+{
+    const RPCArg& arg;
+    const UniValue& value;
+    T defaults;
+    operator T() const;
+};
+
+template <> RPCDefaultParam<int>::operator int() const { return value.isNull() ? defaults : value.get_int(); }
+
+template <typename T>
+struct RPCParam
+{
+    const RPCArg& arg;
+    const UniValue& value;
+    RPCDefaultParam<T> defaults(const T& defaults) const { return {arg, value, defaults}; }
+    operator T() const;
+    bool isNull() const { return value.isNull(); }
+    bool isNum() const { return value.isNum(); }
+    template <typename F>
+    void forEach(F fn) const {
+        if (value.isNull()) return;
+        for (const UniValue& elem : value.get_array().getValues()) {
+            fn(elem);
+        }
+    }
+};
+
+template <> RPCParam<bool>::operator bool() const { return value.get_bool(); }
+template <> RPCParam<int>::operator int() const { return value.get_int(); }
+template <> RPCParam<std::string>::operator std::string() const { return value.get_str(); }
+template <> RPCParam<uint256>::operator uint256() const { return ParseHashV(value, arg.GetName()); }
```

Comment:
6e1fecd39fdadbfd1ea3a2eb3e2c74db43cecb65

This allows to throw with correct parameter name in `ParseHashV`.